lib/detect.js

```diff
@@ -90,33 +74,29 @@ function detectWindows( callback ) {
  * @param {String}   name     Name of a browser
  * @param {Function} callback Callback function
  */
-function checkDarwin( name, callback ) {
-	if ( darwin[ name ] ) {
-		if ( darwin[ name ].all ) {
-			darwin[ name ].all( function( err, available ) {
-				if ( err ) {
-					callback( 'failed to get version for ' + name );
-				} else {
-					callback( err, available );
-				}
-			} );
-		} else {
-			darwin[ name ].version( function( err, version ) {
-				if ( version ) {
-					darwin[ name ].path( function( err, p ) {
-						if ( err ) {
-							return callback( 'failed to get path for ' + name );
-						}
-
-						callback( null, version, p );
-					} );
-				} else {
-					callback( 'failed to get version for ' + name );
-				}
-			} );
-		}
+function checkDarwin(name, callback) {
```

Comment:
Just removed the surrounding `if` statement and moved the manual call of `checkOthers` to the main detection function